src/Tizen.NUI/src/public/BaseComponents/ControlState.cs

```diff
@@ -0,0 +1,315 @@
+                return false;
+
+            return Contains(other);
+        }
+
+        ///  <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object obj) => this.Equals(obj as ControlState);
+
+        ///  <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() => stateList.GetHashCode();
+
+        ///  <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override string ToString()
+        {
+            string name = "";
+            for (int i = 0; i < stateList.Count; i++)
+            {
+                name += ((i == 0) ? "" : ", ") + stateList[i].name;
+            }
+            return name;
+        }
+
+        /// <summary>
+        /// Compares whether the two ControlStates are same or not.
+        /// </summary>
+        /// <param name="lhs">A <see cref="ControlState"/> on the left hand side.</param>
+        /// <param name="rhs">A <see cref="ControlState"/> on the right hand side.</param>
+        /// <returns>true if the ControlStates are equal; otherwise, false.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static bool operator ==(ControlState lhs, ControlState rhs) => lhs.Equals(rhs);
+
+        /// <summary>
+        /// Compares whether the two ControlStates are different or not.
+        /// </summary>
+        /// <param name="lhs">A <see cref="ControlState"/> on the left hand side.</param>
+        /// <param name="rhs">A <see cref="ControlState"/> on the right hand side.</param>
+        /// <returns>true if the ControlStates are not equal; otherwise, false.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static bool operator !=(ControlState lhs, ControlState rhs) => !lhs.Equals(rhs);
+
+        /// <summary>
+        /// The addition operator.
+        /// </summary>
+        /// <param name="lhs">A <see cref="ControlState"/> on the left hand side.</param>
+        /// <param name="rhs">A <see cref="ControlState"/> on the right hand side.</param>
+        /// <returns>The <see cref="ControlState"/> containing the result of the addition.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static ControlState operator +(ControlState lhs, ControlState rhs) => Create(lhs, rhs);
+    }
+
+    /// <summary>
+    /// The Key/Value pair structure. this is mutable to support for xaml.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public struct StateValuePair : IEquatable<StateValuePair>
+    {
+        /// <summary>
+        /// The constructor with the specified state and value.
+        /// </summary>
+        /// <param name="state">The state</param>
+        /// <param name="value">The value associated with state.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public StateValuePair(ControlState state, object value)
```

Comment:
I am wondering if the `Value` is `object` type, what will xaml code would look like,

This is example of Selector<string>, I think it seems not have any problems,
```
<c:TextLabelStyle.Text>
    <c:StateValuePair State="Normal" Value="button"/>
    <c:StateValuePair State="Pressed" Value="pressed!!" />
</c:TextLabelStyle.Text>
```

But how about Selector<Color>?
The Color can be expressed as a string (by Converter), e.g. "Yellow".
```
<c:TextLabelStyle.TextColor>
    <c:StateValuePair State="Normal" Value="Yellow"/>
</c:TextLabelStyle.Text>
```
Is it possible? In my opinion, the StateValuePair would take it as a string (no conversion to Color) and that will cause a problem when getter is called. (e.g. Color a = selector.Normal; -> Can not cast to Color.)

Reply:
Template type T seems much better for `StateValuePair`. it will also support xaml intellisense of `StateValuePair.Value`.